src/netconf_acm.c

```diff
@@ -1535,16 +1569,24 @@ ncac_check_data_read_filter_r(struct lyd_node **first, const char *user)
 void
 ncac_check_data_read_filter(struct lyd_node **data, const char *user)
 {
+    char **groups = NULL;
+    uint32_t group_count;
+
+    if (ncac_collect_groups(sr_get_context(np2srv.sr_conn), user, &groups, &group_count)) {
```

Comment:
Function accesses the global `nacm` structure, needs the lock so it should be moved after it.